### PR TITLE
Retry clicking on delete immediately popup in trashbin

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

This PR handles retrying to click delete popup dialog inside the trashbin page, which has been failing quite a lot.
## Related Issue

https://github.com/owncloud/phoenix/issues/1728

## How Has This Been Tested?

CI
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 1]
```diff
@@ -104,18 +104,21 @@ module.exports = {
...
+      return this
         .waitForFileVisible(fileName)
         .useXpath()
         .performFileAction(fileName, 'delete')
         .waitForElementEnabled(
           this.elements.deleteFileConfirmationBtn.selector,
           'css selector'
         )
-
+        .clickDeleteButton()
```
Comment (jasson99) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: .useCss( )

Comment (HariBhandari07) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: @jasson99 ```clickDeleteButton()``` already returns with ```useCss()```, so attaching ```useCss()``` here makes no sense.

Comment (jasson99) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: does that work in other cases where we need to use deleteFile( ) for other test cases

Comment (jasson99) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: okay I understood now

**Comment by jasson99** [comment]
LGTM

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 2]
```diff
@@ -104,18 +104,21 @@ module.exports = {
...
+    },
+    clickDeleteButton: async function () {
       const clickAction = function () {
-        return this.initAjaxCounters().click('@deleteFileConfirmationBtn')
+        return this.api.initAjaxCounters()
+          .useCss().click(this.elements.deleteFileConfirmationBtn)
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: I think we can make it simpler by
```suggestion
          .click('@deleteFileConfirmationBtn')
```

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 3]
```diff
@@ -104,18 +104,21 @@ module.exports = {
...
         .useXpath()
         .performFileAction(fileName, 'delete')
         .waitForElementEnabled(
           this.elements.deleteFileConfirmationBtn.selector,
           'css selector'
         )
-
+        .clickDeleteButton()
+    },
+    clickDeleteButton: async function () {
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: that function does not click the delete button but confirms the delete action
```suggestion
    confirmDeletion: async function () {
```

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 4]
```diff
@@ -104,18 +104,18 @@ module.exports = {
...
         .useXpath()
         .performFileAction(fileName, 'delete')
-        .waitForElementEnabled(
-          this.elements.deleteFileConfirmationBtn.selector,
-          'css selector'
-        )
-
+        .confirmDeletion()
+    },
+    confirmDeletion: async function () {
```
Comment (kiranparajuli589) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: get rid of `async` if not necessary!

**Comment by kiranparajuli589** [comment]
LGTM :rocket:

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js (hunk 4 above)
Comment (skshetry) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: It's necessary here.

**Comment by HariBhandari07** [comment]
This PR has been tested in this draft PR https://github.com/owncloud/phoenix/pull/2616

**Pull request merged**